src/ripple/app/ledger/Ledger.cpp

```diff
@@ -941,12 +997,16 @@ Ledger::succ (uint256 const& key,
 std::shared_ptr<SLE const>
 Ledger::read (Keylet const& k) const
 {
-    auto const& value =
+    assert(k.key.isNonZero());
```

Comment:
1. Use `== zero`.
2. You can do this in one test:

    if (k.key == zero) 
    {
        assert(false);
        return nullptr;
    }